### PR TITLE
Update release info [4.0.x]

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Release info for the 4.0.x branch.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Review by hzhou — COMMENTED** [review]
LGTM other than the comments on RELEASE_NOTES. I don't mind if we don't mention them since we don't expect them to be common or critical anyway (having incompatible libfabric or ucx or using Solaris).

Separately, do you want to update the libtool version string?

File: RELEASE_NOTES [hunk 1]
```diff
@@ -3,17 +3,7 @@
...
-   - The release is thoroughly tested with embedded libfabric and ucx.
-     Using externally installed libfabric or ucx is likely to succeed
-     but less tested.
-
- * Build Issues
-   - Embedded ucx or libfabric builds will not succeed with Solaris
-     compilers.
-
  * Dynamic process support
-   - Requires fi_tagged capability in OFI netmod
```
[the remark this answers is not in the document]
Comment (raffenet) [reply RELEASE_NOTES:16]: Added back.

**Comment by raffenet** [comment]
> Separately, do you want to update the libtool version string?

Yes, but just the comment for now. The beta/rc will have a libtool string of 0:0:0 by our convention. The GA release will update the age since we have added new interfaces for MPI-4.0.

**Comment by raffenet** [comment]
Updated version string in comment in this push, and addressed comments about the RELEASE_NOTES changes. Should be good to go, and I can start preparing the 4.0b1 tarball.

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**